File: mqtt/src/main/java/org/sheedon/mqtt/Callback.java
package org.sheedon.mqtt;


/**
 * 消息反馈
 * @Author: sheedon
 * @Email: sheedonsun@163.com
 * @Date: 2020/2/11 12:49
 */
public interface Callback<R extends Response> {

    void onFailure(Throwable e);


    void onResponse(R response);
}

File: mqtt/src/main/java/org/sheedon/mqtt/Callback.kt
package org.sheedon.mqtt

import org.sheedon.rr.core.Callback

/**
 * 回调信息
 *
 * @Author: sheedon
 * @Email: sheedonsun@163.com
 * @Date: 2022/1/30 7:57 下午
 */
interface Callback : Callback<Request, Response> {
}